### PR TITLE
District of Columbia (DC) income tax

### LINKED ISSUE #1705: DC income tax
https://code.dccouncil.gov/us/dc/council/code/titles/47/chapters/18/subchapters/VI

### LINKED ISSUE #1708: DC CDCC

### LINKED ISSUE #1710: DC Keep Child Care Affordable Tax Credit
https://otr.cfo.dc.gov/sites/default/files/dc/sites/otr/publication/attachments/2022_D-40_Booklet_Final_blk_01_23_23_Ordc.pdf#page=14

> Schedule ELC, Keep Child Care Affordable Tax Credit.
> This schedule allows eligible taxpayers to claim an early learning
> tax credit for an eligible child under the age of 4 as of 9/30/22 and
> payments made during the taxable year after August 31st if the eligible
> child meets age requirements for enrollment in Pre-K according to DC
> Code §38-273.02(a)

### LINKED ISSUE #1715: DC Homeowner and Renter Property Tax Credit
https://otr.cfo.dc.gov/sites/default/files/dc/sites/otr/publication/attachments/2022_D-40_Booklet_Final_blk_01_23_23_Ordc.pdf#page=48

https://otr.cfo.dc.gov/sites/default/files/dc/sites/otr/publication/attachments/2022_Sch_H.pdf

* Start with property tax or 20% of rent
* Subtract from that amount a percent of AGI (below)
* Limit to $1,250

image

### PR DESCRIPTION
**Pull request by martinholmer**
Fixes #1705 and also fixes #1708 and fixes #1710 and fixes #1715.

The code in this pull request generates the same DC income tax amounts as generated by TAXSIM35 (2023-07-15 patch version) for 800,000 randomly-generated integration tests (for the `p21` sample with DC EITC, and for the `e21` through `k21` samples as described in issue #993).

### Summary
:sparkles::card_file_box::memo:

This pull request adds the District of Columbia (DC) state income tax as a new feature to the policy engine. It creates and modifies several parameter files under `policyengine_us/parameters/gov/states/dc/tax/income` to represent the DC tax code and forms. It also deletes an unused parameter file and updates the changelog.

> _`DC` tax added_
> _New parameters and files_
> _Autumn of updates_

### Walkthrough
*  Add DC state income tax as a new feature and update minor version ([link](https://github.com/PolicyEngine/policyengine-us/pull/2625/files?diff=unified&w=0#diff-5885042762833a4aef51405a4036067e4e6aefa7a5091a2987d1d9d1fc2647e8R1-R4))
*  Add parameter files for DC additions to US AGI, including self-employment losses ([link](https://github.com/PolicyEngine/policyengine-us/pull/2625/files?diff=unified&w=0#diff-a74b37913ec72450226a1fb1317880e3c6c5e402a407915f6185608a17b6ace8R1-R11), [link](https://github.com/PolicyEngine/policyengine-us/pull/2625/files?diff=unified&w=0#diff-6a1b03c53cb667a3a75c69f1b9fcaf5a681a8594729e8a48351bdd5ecc738574R1-R12))
*  Add parameter files for DC non-refundable and refundable credits, including CDCC, EITC, and PTC ([link](https://github.com/PolicyEngine/policyengine-us/pull/2625/files?diff=unified&w=0#diff-10f7d99197620b2b94c1ead554f190d2d641f2675815d5a8c04e64e3f299cd78R1-R12), [link](https://github.com/PolicyEngine/policyengine-us/pull/2625/files?diff=unified&w=0#diff-442dcefc5978b9ba31488224682afff3b424cfe033318a9766cb0653b3b8a896R1-R12), [link](https://github.com/PolicyEngine/policyengine-us/pull/2625/files?diff=unified&w=0#diff-1ebb73829062a04b7067fcc028b40b081c730937837e4346b67b4a8c8a2a29cdR1-R21), [link](https://github.com/PolicyEngine/policyengine-us/pull/2625/files?diff=unified&w=0#diff-fbe39a4186763c71fc19e20f20f3103af8498dae7b56fb03a898923fc09d9e24R1-R29), [link](https://github.com/PolicyEngine/policyengine-us/pull/2625/files?diff=unified&w=0#diff-4ff79a3243aa868142eea635471a092348253610cf0122fb30936776bf33a721R1-R13), [link](https://github.com/PolicyEngine/policyengine-us/pull/2625/files?diff=unified&w=0#diff-7c253bdeb37524f0c7f1a7cf0d8f57a64db6e17c5911093698dd44ff4f9c311aR1-R12), [link](https://github.com/PolicyEngine/policyengine-us/pull/2625/files?diff=unified&w=0#diff-48a12d62f3cd57d56b0a2137bc616828c7bbf8a33772a18e73360729c6022160R1-R12), [link](https://github.com/PolicyEngine/policyengine-us/pull/2625/files?diff=unified&w=0#diff-e719ca18376271099fd0b2d567dc4c94cb255edcae54b1c0bc93320cbbd285c3R1-R13))
*  Modify parameter files for DC EITC for filers without children, adding metadata and values ([link](https://github.com/PolicyEngine/policyengine-us/pull/2625/files?diff=unified&w=0#diff-1d7480fde3ecd7b3b2b4d9a62b7a6ae73bbdff5f9e29019fc2de4ff3a64a519aL2-R13), [link](https://github.com/PolicyEngine/policyengine-us/pull/2625/files?diff=unified&w=0#diff-ea1d662f4161d64b030f64486593213eee4cfa873b3f1facba7ee04f68de8852R1-R15), [link](https://github.com/PolicyEngine/policyengine-us/pull/2625/files?diff=unified&w=0#diff-dc760d0e38aca3231b3d4550fdc83870b0d9a1ac956b00b72ba848f0ded12c6aR1-R15), [link](https://github.com/PolicyEngine/policyengine-us/pull/2625/files?diff=unified&w=0#diff-2e09edeb32a3d8e80e0dbc9e5e8c6a1d6d8f0d44f23603e17266851673e98400L2-R10), [link](https://github.com/PolicyEngine/policyengine-us/pull/2625/files?diff=unified&w=0#diff-607de03beabd0d9d1d3d3734eee00e67b783125a6a5b45d68cb77700527037c8L2-R15))
*  Delete parameter file for DC EITC match rate for filers without children ([link](https://github.com/PolicyEngine/policyengine-us/pull/2625/files?diff=unified&w=0#diff-509cda49fcaf2ca9fe926d4dfaaee99992062553ed5692525a9eb11d4317a56d))
*  Add parameter files for DC itemized deduction phase-out rate and start ([link](https://github.com/PolicyEngine/policyengine-us/pull/2625/files?diff=unified&w=0#diff-c496e489b705a120dfaceb7f101d451dbbb63e9ec1f6322c5b49ba3af456e002R1-R11), [link](https://github.com/PolicyEngine/policyengine-us/pull/2625/files?diff=unified&w=0#diff-001db273638c55157948a53438801e2720144ff7d25c74f56e3f3c541ab45909R1-R22))

### CODE REVIEW

File: policyengine_us/parameters/gov/states/dc/tax/income/subtractions/disabled_exclusion/amount.yaml [hunk 1]
```diff
@@ -0,0 +1,13 @@
+description: DC allows an AGI subtraction of this amount for disabled persons who meet certain eligiblity requirements.
```
Comment (MaxGhenis) [inline policyengine_us/parameters/gov/states/dc/tax/income/subtractions/disabled_exclusion/amount.yaml:1]:
```suggestion
description: DC allows an AGI subtraction of this amount for disabled persons who meet certain eligibility requirements.
```

File: policyengine_us/parameters/gov/states/dc/tax/income/subtractions/disabled_exclusion/income_limit.yaml [hunk 2]
```diff
@@ -0,0 +1,13 @@
+description: DC allows an AGI subtraction for disabled persons who meet certain eligiblity requirements and have household income below this limit
```
Comment (MaxGhenis) [inline policyengine_us/parameters/gov/states/dc/tax/income/subtractions/disabled_exclusion/income_limit.yaml:1]:
```suggestion
description: DC allows an AGI subtraction for disabled persons who meet certain eligibility requirements and have household income below this limit.
```

File: policyengine_us/parameters/gov/states/dc/tax/income/subtractions/sources.yaml [hunk 3]
```diff
@@ -0,0 +1,14 @@
+description: DC subtracts these items US AGI when computing DC AGI.
```
Comment (MaxGhenis) [inline policyengine_us/parameters/gov/states/dc/tax/income/subtractions/sources.yaml:1]:
```suggestion
description: DC subtracts these items from US AGI when computing DC AGI.
```

**Review by MaxGhenis — CHANGES_REQUESTED** [review]
Could you also include the [Keep Child Care Affordable Tax Credit](https://code.dccouncil.gov/us/dc/council/code/sections/47-1806.15)? This is separate from the CDCC. (Does taxsim model this?)

File: policyengine_us/parameters/gov/states/dc/tax/income/deductions/itemized/phase_out/start.yaml [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+  breakdown:
+    - filing_status
+  reference:
+    - title: 2021 DC Form D-40 Booklet, Calculation F
+      href: https://otr.cfo.dc.gov/sites/default/files/dc/sites/otr/publication/attachments/52926_D-40_12.21.21_Final_Rev011122.pdf#page=18
+    - title: 2022 DC Form D-40 Booklet, Calculation F
+      href: https://otr.cfo.dc.gov/sites/default/files/dc/sites/otr/publication/attachments/2022_D-40_Booklet_Final_blk_01_23_23_Ordc.pdf#page=17
+
+SINGLE:
+  2021-01-01:  200_000
```
[the remark this answers is not in the document]
Comment (martinholmer) [reply policyengine_us/parameters/gov/states/dc/tax/income/deductions/itemized/phase_out/start.yaml:14]: Done.

File: policyengine_us/parameters/gov/states/dc/tax/income/subtractions/disabled_exclusion/amount.yaml (hunk 1 above)
Comment (martinholmer) [reply policyengine_us/parameters/gov/states/dc/tax/income/subtractions/disabled_exclusion/amount.yaml:1]: Done.

File: policyengine_us/parameters/gov/states/dc/tax/income/subtractions/sources.yaml (hunk 3 above)
Comment (martinholmer) [reply policyengine_us/parameters/gov/states/dc/tax/income/subtractions/sources.yaml:1]: Done.

File: policyengine_us/parameters/gov/states/dc/tax/income/subtractions/disabled_exclusion/income_limit.yaml (hunk 2 above)
Comment (martinholmer) [reply policyengine_us/parameters/gov/states/dc/tax/income/subtractions/disabled_exclusion/income_limit.yaml:1]: Done.

**Comment by martinholmer** [comment]
@MaxGhenis asked in PR #2625:
> Could you also include the 
[Keep Child Care Affordable Tax Credit](https://code.dccouncil.gov/us/dc/council/code/sections/47-1806.15)?  

So does this refundable credit go to all kids under age 4 (or at most three kids under age 4 as suggested by the form) that have CDCC-eligible expenses (assuming the tax unit meets the DC taxable income eligibility test)?

> This is separate from the CDCC.  

So, in DC can you double dip by getting both 32% of the federal CDCC and the KCCATC?

> Does TAXSIM35 model this?

I don't see it in the Fortran source code.

**Comment by MaxGhenis** [comment]
I'm not familiar with the program but yes I do believe you can claim both the CDCC and KCCATC.

**Comment by martinholmer** [comment]
@MaxGhenis,  For the DC KCCATC how do we determine the care expenses attributable to each young child.   The `tax_unit_childcare_expenses` variable includes care expenses for older children and is a total amount.   The DC KCCATC caps care expenses for each young child.

**Comment by MaxGhenis** [comment]
NYC limits its CDCC to children under age four. Could you do something like our implementation of that? https://github.com/PolicyEngine/policyengine-us/blob/master/policyengine_us/variables/gov/local/ny/nyc/tax/income/credits/cdcc/nyc_cdcc_age_restricted_expenses.py

File: policyengine_us/parameters/gov/states/dc/tax/income/credits/eitc/without_children/phase_in/rate.yaml [hunk 5]
```diff
@@ -0,0 +1,15 @@
+description: DC phases in EITC amount at this rate applied to earned income.
+metadata:
+  unit: /1
+  label: DC EITC phase-in rate for filers without qualifying children
+  period: year
+  reference:
+    - title: Code of the District of Columbia § 47–1806.04 (f)(1)(C)(i)
```
Comment (MaxGhenis) [inline policyengine_us/parameters/gov/states/dc/tax/income/credits/eitc/without_children/phase_in/rate.yaml:7]: I don't see 7.65% in this section. If it refers to the IRC, please apply the relevant parameter from the `irs` folder instead.

File: policyengine_us/tests/policy/baseline/gov/irs/income/taxable_income/adjusted_gross_income/adjusted_gross_income_person.yaml [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+    spm_units:
+      spm_unit:
+        members: [person1, person2, person3]
+    households:
+      household:
+        members: [person1, person2, person3]
+  output:
+    adjusted_gross_income: 90_000
+    # self-employment loss is shared evenly by the head and spouse
+    adjusted_gross_income_person: [55_000, 35_000, 0]
```
Comment (MaxGhenis) [inline policyengine_us/tests/policy/baseline/gov/irs/income/taxable_income/adjusted_gross_income/adjusted_gross_income_person.yaml:81]: Could you add a test case with basic_income and the taxable basic income parameter toggled to true, to pass codecov?

**Review by MaxGhenis — CHANGES_REQUESTED** [review]

File: policyengine_us/parameters/gov/states/dc/tax/income/credits/eitc/without_children/phase_in/rate.yaml (hunk 5 above)
Comment (martinholmer) [reply policyengine_us/parameters/gov/states/dc/tax/income/credits/eitc/without_children/phase_in/rate.yaml:7]: Done.

File: policyengine_us/tests/policy/baseline/gov/irs/income/taxable_income/adjusted_gross_income/adjusted_gross_income_person.yaml (hunk 6 above)
Comment (martinholmer) [reply policyengine_us/tests/policy/baseline/gov/irs/income/taxable_income/adjusted_gross_income/adjusted_gross_income_person.yaml:81]: Done.

**Review by MaxGhenis — APPROVED** [review]

**Pull request merged**